### PR TITLE
Updated to reference Loki's usage of object storage

### PR DESCRIPTION
**Pull request by Alex3k**
**What this PR does / why we need it**:
Being 100% backed by object storage is a massive competitive win for us over tools like Elasticsearch and Splunk. I have updated this page to reference explicitly

### CODE REVIEW

File: docs/sources/fundamentals/overview/_index.md [hunk 1]
```diff
@@ -35,6 +35,11 @@ and allows for efficient query execution.
 
 ## Loki features
 
+
+-  **Log data is stored within Object Store only**
+
+    By leveraging Object Storage as the only data storage mechanism, Loki inherits the reliability and stability of the underlying Object Store. It also capitalizes on both the cost efficiency and operational simplicity of Object Storages over other storage mechanisms like locally attached SSD and HDDs.
```
Comment (JStickler) [inline docs/sources/fundamentals/overview/_index.md:41]:
```suggestion
    By leveraging object storage as the only data storage mechanism, Loki inherits the reliability and stability of the underlying object store. It also capitalizes on both the cost efficiency and operational simplicity of object storage over other storage mechanisms like locally attached solid state drives (SSD) and hard disk drives (HDD).
```

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

**Comment by MasslessParticle** [comment]
@Alex3k can you fix the merge conflict? Then we'll be able to merge

**Comment by GrafanaWriter** [comment]
I believe this can be closed as the content has been updated - see https://grafana.com/docs/loki/latest/storage/

**Comment by JStickler** [comment]
@Alex3k Could you resolve the conflicts on this PR so we can get it merged?

**Comment by JStickler** [comment]
Closing due to inactivity and merge conflicts, replaced by https://github.com/grafana/loki/pull/11443

**Pull request closed without merging**